### ISSUE TITLE
Potential error in training loop

### ISSUE THREAD

**Issue by sholtoarmstrong-iot** [chrischoy/SpatioTemporalSegmentation#46]
In line [train.py:100](https://github.com/chrischoy/SpatioTemporalSegmentation/blob/master/lib/train.py#L100), The "loss.backward()" inside the sub_iter loop. However in line [train.py:68](https://github.com/chrischoy/SpatioTemporalSegmentation/blob/master/lib/train.py#L68) the "optimizer.zero_grad()" is applied outside the loop. 

I am not sure if this is intended as to my knowledge the zero grad should be done between each backwards operation? This means the zero_grad should either be inside the sub_iter loop or the loss.backwards() should be replaced with a batch_loss.backwards() outside of the loop.
